docs/agent-api.asciidoc

```diff
@@ -324,6 +324,14 @@ it's possible to use this option to supply an additional message string that wil
 
 ** `custom` - See <<metadata,metadata section>> for details about this option
 
+** `request` You can associate an error with an instance of `http.IncomingMessage` to gain additional context such as the request url, headers and cookies. However in most cases the agent will detect if an error was in response to a  http request and automatically add the request details for you. See <<http-requests, http requests section>> for more details.
```

Comment:
Besides adding a few line-breaks to adhere to our [One Sentence Per Line](https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line) rule from the AsciiDoc Recommended Practices, I took the liberty to move the information about `http.IncomingMessage` to the beginning of the line so it's similar to the rest of the API docs ☺️

This, however, requires that you add the following line to the `docs/index.asciidoc` file:

```asciidoc
:type-incomingmessage: https://nodejs.org/api/http.html#http_class_http_incomingmessage[<http.IncomingMessage>]
```

Here's the complete change:

```suggestion
** `request` +{type-incomingmessage}+ You can associate an error with information about the incoming request to gain additional context such as the request url, headers, and cookies.
However, in most cases, the agent will detect if an error was in response to an http request and automatically add the request details for you.
See <<http-requests,http requests section>> for more details.
```